### ISSUE TITLE
Subtitle asset loading problem 

### ISSUE THREAD

**Issue by JamesTharakan** [Dash-Industry-Forum/livesim2#207]
The subtitle in vtt format is packaged with shakaPackager, below is the generated mpd segment. When loading in livesim2 we are getting the below error message. 

```
   <AdaptationSet id="2" contentType="text" lang="en" segmentAlignment="true">
      <Role schemeIdUri="urn:mpeg:dash:role:2011" value="subtitle"/>
      <Representation id="2" bandwidth="990" codecs="stpp" mimeType="application/mp4">
      </Representation>
        <SegmentTemplate timescale="1000" initialization="subtitles_1/text_init.mp4" media="subtitles_1/$Number$.m4s" startNumber="1">
        </SegmentTemplate>	  
    </AdaptationSet>
```

> level=WARN msg="Asset loading problem. Skipping" asset=11/manifest.mpd err="getRep: addRegExpAndInit: decode init: decode box moov: decode box trak: decode box mdia: decode box minf: decode box stbl: decode box stsd: child stpp size mismatch in stsd: 44 - 43"

**Comment by tobbee** [comment]
Hi @JamesTharakan,

livesim2 does not yet support wvtt subtitles. Issue #46 is about adding support for sources with wvtt segments. However, your content is not VTT but STPP (TTML) according to the manifest.

`stpp` should be supported. I suspect this is an issue about a zero-byte end character in a utf8list in the stpp box as decoded by mp4ff. Could you upload the init segment of the asset as an attachment? Ideally with a non-mp4 file extension like ".init" to avoid having Github convert the link into a player link.

**Comment by JamesTharakan** [comment]
![text_init](https://github.com/user-attachments/assets/10a8ee9a-e9a6-419b-a2bd-671a0361eeec)
Attached the text_init.mp4 as text_init.jpg

**Comment by JamesTharakan** [comment]
![text_init](https://github.com/user-attachments/assets/ed403b3e-017a-491a-8beb-d8aa65379330)
Update the text_init.mp4

**Comment by JamesTharakan** [comment]
$ MP4Box -info 1.m4s
[iso file] Movie fragment but no moov (yet) - possibly broken parsing!
File is a segment - 1 movie fragments - Brand mp41 (version 0):
        Fragment #1 size 377 (moof 104) {TKID 1 TFDT 0}



$ MP4Box -info text_init.mp4
 Movie Info - 1 track - TimeScale 1000
Duration 00:00:00.000
Fragmented: yes - duration 02:09:52.000
0 fragments - 0 SegmentIndexes
Progressive (moov before mdat)
Major Brand mp41 - version 0 - compatible brands: iso8 isom mp41 dash cmfc
Created: GMT Fri Jul 26 06:24:55 2024

 Movie Meta type: "ID32" - 0 resource item(s)


 Track 1 Info - ID 1 - TimeScale 1000
Media Duration 00:00:00.000
Track flags: Enabled In Movie In Preview
Media Language: English (eng)
Media Samples: 0
Fragmented track: 0 samples - Media Duration 00:00:00.000
Fragment sample defaults: duration 8000 size 0 stsd 1 sync 1 padding 0 degradation_priority 0
Media Type: subt:stpp
        XML Subtitle Stream - namespace http://www.w3.org/ns/ttml
        Size 0 x 0 - Translation X=0 Y=0 - Layer 0
        RFC6381 Codec Parameters: stpp.ttml
        All samples are sync
        Max sample duration: 0 / 1000

**Comment by tobbee** [comment]
I looked at the file with mp4ff-info and it works fine. The root problem with stpp size mismatch goes back to https://github.com/Eyevinn/mp4ff/issues/228

It was solved in v0.43 of mp4ff which was included in livesim2 v1.3.0. Which version of livesim2 do you use?

**Comment by JamesTharakan** [comment]
Thanks, once I took the latest version, problem is solved.
